### ISSUE TITLE
A question

### ISSUE THREAD

**Issue by francqz31** [BUTSpeechFIT/TS-ASR-Whisper#1]
amazing work , I wanted to ask so does this work like diarization ? for example can I upload 10 hours of audio and choose a certain speaker by giving in reference audio, and then TS-ASR-Whisper would extract the parts where that certain speaker is talking ?

**Comment by domklement** [comment]
Hello @francqz31,

Thank you! This work proposes a simple method of turning the Whisper (single-talker ASR model) to target speaker ASR model by conditioning the encoder on diarization information. 

So, as long as you have the 10hours of audio diarized, you can pass the diarization output along with the 10h audio to our method and run the inference to get the target speaker transcript. However, the accuracy of produced transcript will depend on the diarization accuracy.

We're currently polishing the code and training some models on larger datasets. We will release the pretrained weights, so you will be ale to play with the model soon.

Let us know if you have any further question.

Best,
Dominik

**Comment by francqz31** [comment]
Thank you so much for the detailed answer, one last question which diarization algorithm do you recommend to work best with your paper? or what is the current SOTA in your opinion?

**Comment by domklement** [comment]
No worries. Glad to help.

**Short answer:**
1. If you're not experienced in diarization, use [pyannote diarization 3.1](https://huggingface.co/pyannote/speaker-diarization-3.1) and perhaps play with the clustering hyperparameters to tune the performance on your dataset (do not measure DER only, but decompose it to FA, Miss, Confusion). Optionally, if you have some subset that is annotated, fine-tune pyannote pipeline as it's going mitigate the domain mismatch between your data and the training data. 
2. If you are experienced in diarization, then train something that works the best on your target dataset or try more advanced public diarization pipelines, but I'd use pyannote 3.1 as a baseline either way. There're also various approaches to fusing multiple diarization systems together and also postprocessing techniques, such as median filtering of the produced speaker labels (to smooth out the probabilities or directly the predictions).

**Longer, more elaborated answer:**
Regarding diarization, it's hard to tell and it heavily depends on the target dataset. Also, it may depend on what errors you are willing to sacrifice (FA, Miss, Confusion). I'd say all of the mentioned types of diarization errors can cause different subsequent transcription errors. Given that our method heavily depends on the diarization, false alarms (i.e. there's silence but the diar. system says it's some speaker) are the least harmful as the Whisper itself can cope with silence to some extent. On the other hand, speaker confusion or misses (in some cases) can harm the system quite a bit. In general, I'd say confusion is quite important but imo there's no way you can balance confusion with other type of errror (as opposed to FA/Miss for example where you can trade one in favour of the other).

We're currently deeply exploring different effects of diarization errors on the transcription, but I'd say, first, select and tune some diarization pipeline (e.g. aforementioned pyannote) and run TS-ASR. Observe the mistakes it makes and try to re-tune diarization to correct the produced mistakes and iterate. That's the best I can advice you to do right now. Also, you can fine-tune TS-ASR (once we release the codebase) on your target dataset.

Best,
Dominik

**Comment by francqz31** [comment]
Thank you so much , I now have a better understanding of the paper and the whole approach and I will try what you said above.
If you don't mind since your whole field of research is Diarization , I would like to share with you a small gemstone that I discovered early this year , there is a researcher that created a new approach for Diarization, tbh it is the most accurate one that I ever seen it makes pyannote look silly , the outcome results feels like someone manually annotated it. It is this repo here https://github.com/KoljaB/WhoSpeaks . if you are interested you can read the code and follow a similar approach or create a new idea out of it.
Best,
Yours.

**Comment by domklement** [comment]
Thank you, will definitely check it out. 

If you have any further question, feel free to reopen this issue. For now, I'm closing it.

Best,
Dominik
